### DISCUSSION TITLE
Add "smart" log position

### DISCUSSION THREAD

**Discussion by Goose97** [Goose97/timber.nvim#7] [Ideas]
We support two main log positions `above` and `below`. However, most of the time, there is a more favorable. For example, in variable declarations, we always want to put the log below:
```lua
local foo = "foo"
print("foo", foo)
```

Whereas, in return statements, we always want to put the log above:
```lua
print("foo", foo)
print("bar", bar)
return foo + bar
```

This idea proposes a new log placement called `smart`. We will hand pick the position for every scenario. Implementation wise, this is quite easy, we just need to mark a "prefer" logable range in the Treesitter query. However, the difficult part is coming up with a sensible default. If there's no sensible default, can we let users customize it?

I don't see an easy way to customize without requiring users to write Treesitter queries. This is too much for a lot of average users.

**Comment by karamanliev** [comment]
Until now I was using [logsitter.nvim](https://github.com/gaelph/logsitter.nvim) for inserting debug statements. I use it for ts/lua/python and I think that generally it is working nice. You could check what treesitter nodes he uses for the smart placement for every supported language [here](https://github.com/gaelph/logsitter.nvim/tree/main/lua/logsitter/lang). In my opinion, there's no shame in copying good implementations/ideas when you offer other unique features.
